### PR TITLE
Add citation file for Reshma's work and resulting thesis.

### PR DESCRIPTION
**Pull request by egrace479**
Adding `CITATION.cff` for Reshma's work with a preferred-citation to include her thesis resulting from it. The `<commit>` (line 15) for the release will be the one with this addition to the main branch, and will thus be filled in after merging and releasing to clearly version Reshma's work. This PR will be only 1 commit, so if changes are needed I will squash all into a single commit before the merge (with branch protections this will require an approval following the final push).

@reshmarameshbabu I intend to this on Monday (Dec. 4th), so please let us know if you have any concerns about this prior to that date. I have used the keywords you input for your thesis and got your ORCID from the [OhioLINK ETD entry](http://rave.ohiolink.edu/etdc/view?acc_num=osu168198420667979).

### CODE REVIEW

**Review by thompsonmj — DISMISSED** [review]
LGTM

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    value: "https://github.com/Imageomics/Butterfly-mimicry/tree/<commit>"
+keywords:
+  - "imageomics"
+  - "butterflies"
+  - "mimicry"
+  - "machine learning"
+  - "computer vision"
+  - "traits"
+license: MIT
+message: "If you use this software, please cite both the thesis from preferred-citation and the software itself."
```
Comment (hlapp) [inline CITATION.cff:24]: Do you know that the preferred citation will be shown as "preferred-citation"? Perhaps we will only know once merged.

**Review by hlapp — DISMISSED** [review]
LGTM, see inline comment about the dash in "preferred-citation".

File: CITATION.cff (hunk 1 above)
Comment (egrace479) [reply CITATION.cff:24]: The displayed citation is the preferred citation:
Screenshot 2023-12-04 at 10 01 55 AM

Comment (hlapp) [reply CITATION.cff:24]: So it doesn't seem like it's using the `message` in the display, am I right?

Comment (egrace479) [reply CITATION.cff:24]: Yes, it doesn't seem so, and I'm not sure why, as that is the purpose of the [`message`](https://github.com/citation-file-format/citation-file-format/blob/main/schema-guide.md#message) parameter.

**Comment by egrace479** [comment]
I updated the release date to today, so I'll do the release as soon as this is merged.

**Review by thompsonmj — APPROVED** [review]
🚢

**Pull request merged**